src/utils/feedData.js

```diff
@@ -66,6 +79,42 @@ export function mapTransactionsHistory(
         ...rest,
       };
     });
+
+  if (keepHashDuplicatesIfBetweenAccounts) {
+    const accountsAddresses = accounts.map((acc) => getAccountAddress(acc));
+    const ascendingHistory = orderBy(concatedHistory, ['createdAt'], ['asc']);
+
+    const historyWithTrxBetweenAcc = ascendingHistory.reduce((alteredHistory, historyItem) => {
+      const { from: fromAddress, to: toAddress, hash } = historyItem;
+      if (alteredHistory.some((item) => item.hash === hash)) {
```

Comment:
IMO if we extract all those .some into some descriptive variables, the code will become more readable